### PR TITLE
Feature/tg 140/mentor again

### PR DESCRIPTION
**Pull request by victor-paumier**

### CODE REVIEW

File: src/Controller/DashboardController.php [hunk 1]
```diff
@@ -102,7 +102,11 @@ public function reasonRefusalMatching(ObjectManager $om,
     /**
      * @Route("/buddy/{id}/report_mentor", name="matching_report_mentor", methods={"GET", "POST"}, options={ "expose" : true }, condition="request.isXmlHttpRequest()")
      */
-    public function reportMentor(ObjectManager $om, MailManager $mailManager, TranslatorInterface $translator, Request $request, Buddy $buddy)
+    public function reportMentor(ObjectManager $om,
```
Comment (JeremieSamson) [inline src/Controller/DashboardController.php]:
```suggestion
    public function reportMentor(
        ObjectManager $om,
```

File: src/Controller/DashboardController.php [hunk 2]
```diff
@@ -136,4 +140,32 @@ public function reportMentor(ObjectManager $om, MailManager $mailManager, Transl
             'buddy' => $buddy,
         ]);
     }
+
+    /**
+     * @Route("/mentor_next_semester/{response}", name="mentor_next_semester")
+     */
+    public function answerNextSemester(ObjectManager $om, UserInterface $user, TranslatorInterface $translator, $response)
```
Comment (JeremieSamson) [inline src/Controller/DashboardController.php]: multiline

File: src/Controller/DashboardController.php [hunk 3]
```diff
@@ -136,4 +140,32 @@ public function reportMentor(ObjectManager $om, MailManager $mailManager, Transl
             'buddy' => $buddy,
         ]);
     }
+
+    /**
+     * @Route("/mentor_next_semester/{response}", name="mentor_next_semester")
```
Comment (JeremieSamson) [inline src/Controller/DashboardController.php]: add requirement on `{response}` plz

File: src/Controller/HobbyController.php [hunk 4]
```diff
@@ -168,6 +168,7 @@ public function translate(ObjectManager $om,
      */
     public function delete(ObjectManager $om, Request $request, Hobby $hobby): Response
     {
+        $em = $this->getDoctrine()->getManager();
```
Comment (JeremieSamson) [inline src/Controller/HobbyController.php]: why ?

File: src/Controller/MotivationController.php [hunk 5]
```diff
@@ -192,6 +192,7 @@ public function translate(ObjectManager $om,
      */
     public function delete(ObjectManager $om, Request $request, Motivation $motivation): Response
     {
+        $em = $this->getDoctrine()->getManager();
```
Comment (JeremieSamson) [inline src/Controller/MotivationController.php]: why ?

File: src/Controller/UserController.php [hunk 6]
```diff
@@ -59,13 +62,14 @@ public function index(Request $request,
     public function getListUsers(Request $request,
```
Comment (JeremieSamson) [inline src/Controller/UserController.php]:
```suggestion
    public function getListUsers(
        Request $request,
```

File: src/Controller/UserController.php [hunk 7]
```diff
@@ -59,13 +62,14 @@ public function index(Request $request,
...
                                  UserInterface $user,
                                  UserRepository $userRepo,
-                                 TranslatorInterface $translator): JsonResponse
+                                 TranslatorInterface $translator,
+                                 AccessDecisionManagerInterface $decisionManager): JsonResponse
```
Comment (JeremieSamson) [inline src/Controller/UserController.php]:
```suggestion
                                 AccessDecisionManagerInterface $decisionManager
                             ): JsonResponse {
```

File: src/Controller/UserController.php [hunk 8]
```diff
@@ -59,13 +62,14 @@ public function index(Request $request,
...
     {
```
Comment (JeremieSamson) [inline src/Controller/UserController.php]:
```suggestion

```

File: src/Controller/UserController.php [hunk 9]
```diff
@@ -78,7 +82,7 @@ public function getListUsers(Request $request,
             $columnName = $column['data'];
 
             $searchValue = $column['search']['value'];
-            if (!empty($searchValue)) {
+            if ('' !== $searchValue) {
```
Comment (JeremieSamson) [inline src/Controller/UserController.php]:
```suggestion
            if (!empty($searchValue)) {
```

File: src/Entity/User.php [hunk 10]
```diff
@@ -286,6 +286,13 @@ class User extends BaseUser
      */
     private $reports;
 
+    /**
+     * @var bool
+     *
+     * @ORM\Column(type="boolean", nullable=false)
+     */
+    private $askNextSemester = false;
```
Comment (JeremieSamson) [inline src/Entity/User.php]: Maybe `$hasBeenAskedForNextSemester` ?

Comment (JeremieSamson) [reply src/Entity/User.php]: or `$askedForNextSemester`

File: src/Controller/UserController.php (hunk 9 above)
Comment (victor-paumier) [reply src/Controller/UserController.php]: I have some $searchValue == 0, doesn't works with empty

File: src/Controller/HobbyController.php (hunk 4 above)
Comment (victor-paumier) [reply src/Controller/HobbyController.php]: bad merge

File: src/Controller/MotivationController.php (hunk 5 above)
Comment (victor-paumier) [reply src/Controller/MotivationController.php]: bad merge

File: src/Controller/DashboardController.php (hunk 1 above)
Comment (victor-paumier) [reply src/Controller/DashboardController.php]: done

File: src/Controller/DashboardController.php (hunk 2 above)
Comment (victor-paumier) [reply src/Controller/DashboardController.php]: done

File: src/Controller/DashboardController.php (hunk 3 above)
Comment (victor-paumier) [reply src/Controller/DashboardController.php]: done

File: src/Controller/UserController.php (hunk 6 above)
Comment (victor-paumier) [reply src/Controller/UserController.php]: done

File: src/Controller/UserController.php (hunk 7 above)
Comment (victor-paumier) [reply src/Controller/UserController.php]: done

File: src/Controller/UserController.php (hunk 8 above)
Comment (victor-paumier) [reply src/Controller/UserController.php]: done

**Pull request merged**